### ISSUE TITLE
global-require false positive

### ISSUE THREAD

**Issue by graingert** [eslint/eslint#6678]
**What version of ESLint are you using?**
v3.0.1
**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**
```json
{
    "extends": "airbnb-base/legacy",
    "parser": "babel-eslint",
    "rules": {
        "max-len": [2, 211, 2, {
          "ignoreUrls": true,
          "ignoreComments": false
        }],
        "no-unused-vars": 0,
        "func-names": 0,
        "vars-on-top": 0,
        "no-param-reassign": 0,
        "eqeqeq": 0,
        "guard-for-in": 0,
        "no-use-before-define": 0,
        "no-underscore-dangle": 0,
        "no-restricted-syntax": [
            2,
            "DebuggerStatement",
            "LabeledStatement",
            "WithStatement"
        ],
        "comma-dangle": 0
    },
    "settings": {
        "import/resolver": "webpack"
    }
}
```
**What did you do? Please include the actual source code causing the issue.**
var apiUrl = require('../config').apiUrl + '/api/v1/';
**What did you expect to happen?**
should be valid under global-require rules.
**What actually happened? Please include the actual, raw output from ESLint.**
[1] /redacted/file.js
[1]   1:14  error  Unexpected require()  global-require
[1] 
[1] ✖ 1 problem (1 error, 0 warnings)

**Comment by ilyavolodin** [comment]
Could you provide source code of the file you are linting? Is require the only line of code in that file?

**Comment by graingert** [comment]
@ilyavolodin it still fails with just that line

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed.

![image](https://cloud.githubusercontent.com/assets/1937871/16879226/a07c507a-4aea-11e6-9436-ea7f51e0c833.png)

But I guess probably this is an intentional behavior.
As mentioned in a previous discussion (#5773), `global-require` is the rule which emulates ES Module's `import` for `require`. Then we cannot use binary expressions in `import` declarations.

**Comment by graingert** [comment]
but it allows require('x').default();

**Comment by nzakas** [comment]
I agree with @mysticatea, I believe this rule is behaving as expected.

**Comment by graingert** [comment]
@mysticatea @nzakas what's the difference between calling a method and adding a string?

What operations are allowed on a required object?

**Comment by nzakas** [comment]
@graingert it's easier if you take a look at the constructed AST in each case. Plug them into astexplorer.net and you should be able to spot the difference.

**Comment by graingert** [comment]
@nzakas they are clearly different tree names, but there doesn't seem to be a reason why one is allowed and other is not.

```import x from 'foo'.default();```
does not work just as much as
```(import x from 'foo') + '3';

**Comment by nzakas** [comment]
IIRC, the `require("foo").default()` case was allowed because it was pretty common, but I'm not 100% sure. It doesn't really matter in the evaluation of this proposal, though, as we can't really remove existing functionality that people might be relying on whereas we can decide not to add new functionality without paying a cost.

I'm of the mind that we shouldn't make any change here, but I'll leave open to see if any other team members feel strongly otherwise.

**Comment by kaicataldo** [comment]
I'm closing this issues because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
